afs-storage-api/src/test/java/com/powsybl/afs/storage/AbstractAppStorageTest.java

```diff
@@ -10,594 +10,1011 @@
+        // create 2 folders
+        NodeInfo folder1Info = storage.createNode(test34FolderInfo.getId(), "test3", FOLDER_PSEUDO_CLASS, "", 0, new NodeGenericMetadata());
+        NodeInfo folder2Info = storage.createNode(test34FolderInfo.getId(), "test4", FOLDER_PSEUDO_CLASS, "", 0, new NodeGenericMetadata());
+        storage.setConsistent(folder1Info.getId());
+        storage.setConsistent(folder2Info.getId());
         storage.flush();
 
-        discardEvents(2);
-
-        storage.addDependency(folder3Info.getId(), "dep", folder4Info.getId());
-        storage.addDependency(folder3Info.getId(), "dep2", folder4Info.getId());
+        // Add the dependency
+        storage.addDependency(folder1Info.getId(), "dep", folder2Info.getId());
+        storage.addDependency(folder1Info.getId(), "dep2", folder2Info.getId());
         storage.flush();
 
-        discardEvents(2);
+        // Check that the dependencies exist
+        assertEquals(Collections.singleton(folder2Info), storage.getDependencies(folder1Info.getId(), "dep"));
+        assertEquals(Collections.singleton(folder2Info), storage.getDependencies(folder1Info.getId(), "dep2"));
 
-        assertEquals(Collections.singleton(folder4Info), storage.getDependencies(folder3Info.getId(), "dep"));
-        assertEquals(Collections.singleton(folder4Info), storage.getDependencies(folder3Info.getId(), "dep2"));
+        // Delete the node 4
+        storage.deleteNode(folder2Info.getId());
 
-        storage.deleteNode(folder4Info.getId());
-
-        assertTrue(storage.getDependencies(folder3Info.getId(), "dep").isEmpty());
-        assertTrue(storage.getDependencies(folder3Info.getId(), "dep2").isEmpty());
+        // Check that the dependencies do not exist anymore
+        assertTrue(storage.getDependencies(folder1Info.getId(), "dep").isEmpty());
+        assertTrue(storage.getDependencies(folder1Info.getId(), "dep2").isEmpty());
     }
 
-    private void assertRenameNode(NodeInfo rootFolderInfo) {
-        NodeGenericMetadata folder5Metadata = new NodeGenericMetadata().setString("k", "v");
-        NodeInfo folder5Info = storage.createNode(rootFolderInfo.getId(), "test5", FOLDER_PSEUDO_CLASS, "", 0, folder5Metadata);
-        storage.setConsistent(folder5Info.getId());
-        NodeInfo folder51Info = storage.createNode(folder5Info.getId(), "child_of_test5", FOLDER_PSEUDO_CLASS, "", 0, new NodeGenericMetadata());
-        NodeInfo folder52Info = storage.createNode(folder5Info.getId(), "another_child_of_test5", FOLDER_PSEUDO_CLASS, "", 0, new NodeGenericMetadata());
-        storage.setConsistent(folder51Info.getId());
-        storage.setConsistent(folder52Info.getId());
-        storage.flush();
+    @Test
+    @Order(35)
+    public void renameNodeTest() {
+        // Get the node
+        NodeInfo folder5Info = storage.getChildNode(test35FolderInfo.getId(), "test5").orElseThrow(AssertionError::new);
 
+        // Rename the parent
         String newName = "newtest5";
```

Comment:
You can use `"newtest5" ` directly in the `renameNode(...)` call of the next line and remove this variable.